cirq-core/cirq/experiments/xeb_fitting.py

```diff
@@ -66,18 +66,37 @@ def benchmark_2q_xeb_fidelities(
          sampled_df: The sampled results to benchmark. This is likely produced by a call to
             `sample_2q_xeb_circuits`.
         circuits: The library of circuits corresponding to the sampled results in `sampled_df`.
-        cycle_depths: The sequence of cycle depths to simulate the circuits.
+        cycle_depths: The sequence of cycle depths to benchmark the circuits. If not provided,
+            we use the cycle depths found in `sampled_df`. All requested `cycle_depths` must be
+            present in `sampled_df`.
         param_resolver: If circuits contain parameters, resolve according to this ParamResolver
             prior to simulation
         pool: If provided, execute the simulations in parallel.
 
     Returns:
         A DataFrame with columns 'cycle_depth' and 'fidelity'.
     """
+    sampled_cycle_depths = (
+        sampled_df.index.get_level_values('cycle_depth').drop_duplicates().sort_values()
+    )
+    if cycle_depths is not None:
+        if len(cycle_depths) == 0:
+            raise ValueError("`cycle_depths` should be a non-empty array_like")
+        not_in_sampled = np.setdiff1d(cycle_depths, sampled_cycle_depths)
+        if len(not_in_sampled) > 0:
```

Comment:
Just `if not_in_sampled:`?

Reply:
I know some style guides disagree, but I think in Cirq and me personally we usually are explicit. I think it reads closer to the intent